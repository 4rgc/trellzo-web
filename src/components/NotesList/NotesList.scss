.notes-list {
	display: flex;
	flex-direction: column;
	padding: 5px;
	background-color: #9191a1;
	width: 200px;
	border-radius: 2px;
	min-height: 0;
	.card {
		width: 200px;
		margin-bottom: 5px;
	}
	.card:last-child {
		margin-bottom: 0;
	}
	.notes-list-name {
		font-weight: 300;
		text-align: left;
		margin-bottom: 5px;
		margin-left: 10px;
	}
	.notes-list-empty-msg {
		color: white;
		font-weight: 200;
		margin: 10px;
	}
}
