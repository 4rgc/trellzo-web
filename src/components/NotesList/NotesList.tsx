import { Droppable } from '@hello-pangea/dnd';
import { FC } from 'react';
import List from '../../types/List';
import NoteCard from '../NoteCard';
import './NotesList.scss';

export type NotesListProps = {
	list: List;
};

const NotesList: FC<NotesListProps> = ({ list }) => {
	return (
		<Droppable droppableId={list._id}>
			{(provided) => {
				return (
					<div
						className="notes-list"
						{...provided.droppableProps}
						ref={provided.innerRef}
					>
						<span className="notes-list-name">{list.name}</span>
						{list.notes.length ? (
							<>
								{list.notes
									.sort(
										(a, b) =>
											list.notesOrder.indexOf(a._id) -
											list.notesOrder.indexOf(b._id)
									)
									.map((n, index) => (
										<NoteCard
											key={n._id}
											note={n}
											index={index}
										/>
									))}
								{provided.placeholder}
							</>
						) : (
							<span className="notes-list-empty-msg">
								Nothing here yet.
							</span>
						)}
					</div>
				);
			}}
		</Droppable>
	);
};

export default NotesList;
