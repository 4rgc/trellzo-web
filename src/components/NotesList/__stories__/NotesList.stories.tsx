import { DragDropContext } from '@hello-pangea/dnd';
import { Meta, StoryFn } from '@storybook/react';
import NotesList from '../NotesList';

export default {
	title: 'Components/NotesList',
	args: {
		list: {
			_id: '60c5d161a6c781e43c11a5a3',
			name: 'List 1',
			notes: [
				{
					_id: '60c5d167a6c781e43c11a5a4',
					name: 'Note Something',
					description: 'Buy something',
					tags: [],
				},
				{
					_id: '6106f35927d5f4001d197e65',
					name: 'Delete me',
					description: '',
					tags: [],
				},
			],
			notesOrder: [
				'60c5d167a6c781e43c11a5a4',
				'6106f35927d5f4001d197e65',
			],
			boardId: '60c9cdae9c9ae6133cdcd03b',
		},
	},
} as Meta<typeof NotesList>;

const Template: StoryFn<typeof NotesList> = (args) => (
	<DragDropContext onDragEnd={() => {}}>
		<NotesList {...args} />
	</DragDropContext>
);

export const Default = Template.bind({});

export const Empty = Template.bind({});
Empty.args = {
	list: {
		_id: '1',
		name: 'List aa',
		notes: [],
		notesOrder: [],
		boardId: '2',
	},
};
