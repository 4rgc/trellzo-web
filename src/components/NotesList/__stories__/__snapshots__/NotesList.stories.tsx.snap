// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/NotesList Default smoke-test 1`] = `
<div class="notes-list"
     data-rfd-droppable-id="60c5d161a6c781e43c11a5a3"
     data-rfd-droppable-context-id=":r0:"
>
  <span class="notes-list-name">
    List 1
  </span>
  <div class="card card-sm card-noimage note"
       data-rfd-draggable-context-id=":r0:"
       data-rfd-draggable-id="60c5d167a6c781e43c11a5a4"
       tabindex="0"
       role="button"
       aria-describedby="rfd-hidden-text-:r0:-hidden-text-:r1:"
       data-rfd-drag-handle-draggable-id="60c5d167a6c781e43c11a5a4"
       data-rfd-drag-handle-context-id=":r0:"
       draggable="false"
  >
    <div class="card-container-no-text">
      <h3 class="card-title">
        Note Something
      </h3>
      <div style="display: flex; flex-direction: column; flex-grow: 1;">
        <div class="note-card-description card-content-text">
          Buy something
        </div>
      </div>
    </div>
  </div>
  <div class="card card-sm card-noimage note"
       data-rfd-draggable-context-id=":r0:"
       data-rfd-draggable-id="6106f35927d5f4001d197e65"
       tabindex="0"
       role="button"
       aria-describedby="rfd-hidden-text-:r0:-hidden-text-:r1:"
       data-rfd-drag-handle-draggable-id="6106f35927d5f4001d197e65"
       data-rfd-drag-handle-context-id=":r0:"
       draggable="false"
  >
    <div class="card-container-no-text">
      <h3 class="card-title">
        Delete me
      </h3>
      <div style="display: flex; flex-direction: column; flex-grow: 1;">
        <div class="note-card-description card-content-text">
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components/NotesList Empty smoke-test 1`] = `
<div class="notes-list"
     data-rfd-droppable-id="1"
     data-rfd-droppable-context-id=":r5:"
>
  <span class="notes-list-name">
    List aa
  </span>
  <span class="notes-list-empty-msg">
    Nothing here yet.
  </span>
</div>
`;
